VideoWeb/VideoWeb/Extensions/ConferenceEventRequestExtensions.cs

```diff
@@ -24,43 +32,51 @@ public static bool IsParticipantAndVmrEvent(this ConferenceEventRequest request)
-                    EventType.Transfer when !participantEventRequest.ParticipantRoomId.IsNullOrEmpty() => EventType
-                        .RoomParticipantTransfer,
-                    _ => participantEventRequest.EventType
-                };
-
-                return participantEventRequest;
-            }).ToList();
+                    var json = JsonConvert.SerializeObject(request);
+                    var participantEventRequest = JsonConvert.DeserializeObject<ConferenceEventRequest>(json);
+                    participantEventRequest.ParticipantId = p.ToString();
+                    participantEventRequest.ParticipantRoomId = roomId.ToString();
+                    participantEventRequest.EventType = request.EventType switch
+                    {
+                        EventType.Joined when !participantEventRequest.ParticipantRoomId.IsNullOrEmpty() => EventType
+                            .RoomParticipantJoined,
+                        EventType.Disconnected when !participantEventRequest.ParticipantRoomId.IsNullOrEmpty() =>
+                            EventType
+                                .RoomParticipantDisconnected,
+                        EventType.Transfer when !participantEventRequest.ParticipantRoomId.IsNullOrEmpty() => EventType
+                            .RoomParticipantTransfer,
+                        _ => participantEventRequest.EventType
+                    };
+                    
+                    return participantEventRequest;
+                })
+                .ToList();
         }
 
         public static ConferenceEventRequest UpdateEventTypeForVideoApi(this ConferenceEventRequest request)
         {
             request.EventType = request.EventType switch
             {
-                EventType.Joined when !request.ParticipantRoomId.IsNullOrEmpty() => EventType
-                    .RoomParticipantJoined,
+                EventType.Joined when !request.ParticipantRoomId.IsNullOrEmpty() => EventType.RoomParticipantJoined,
                 EventType.Disconnected when !request.ParticipantRoomId.IsNullOrEmpty() => EventType
                     .RoomParticipantDisconnected,
-                EventType.Transfer when !request.ParticipantRoomId.IsNullOrEmpty() => EventType
-                    .RoomParticipantTransfer,
+                EventType.Transfer when !request.ParticipantRoomId.IsNullOrEmpty() => EventType.RoomParticipantTransfer,
                 _ => request.EventType
             };
             return request;
         }
+
+        public static void UpdateEventsTypeForVmrParticipants(this ConferenceEventRequest request,
+            Conference conference)
+        {
+            if (request.ParticipantRoomId.IsNullOrEmpty() || !conference.IsConferenceInSession() ||
+                !request.IsParticipantInVmr(conference)) return;
+            request.EventType = EventType.RoomParticipantTransfer;
+            request.TransferTo = nameof(RoomType.HearingRoom);
+        }
```

Comment:
I don't think we should change the event type of a request like this. Could we get these details in the JoinedEventHandler?

Reply:
No, there is nothing we can use from the JoinedEvent,

Reply:
I see. Could this logic be part of CreateEventsForParticipantsInRoom instead then? We could generate 2 events for a user that's joining if the conference is in session instead.

Reply:
I misread the if statement that calls CreateEventsForParticipantsInRoom. This wouldn't work. I still don't think we should change the request event type and should either have multiple events or have the handler be able to determine where the vmr is and set the participant status based on that.

I also don't think this will work if the JOH's are in a consultation.